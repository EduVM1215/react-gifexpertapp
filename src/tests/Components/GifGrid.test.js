import React from 'react';
import '@testing-library/jest-dom'
import {shallow} from 'enzyme'
import {GifGrid} from '../../Components/GifGrid'
import { useFetchGifs } from '../../Hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs')

describe('Tests sobre GifGrid', () => { 


    const category = 'Metal Gear'
    //const wrapper = shallow(<GifGrid category = {category}/>)

    test('debe de mostrar el componente correctamente', () => { 

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        })
        const wrapper = shallow(<GifGrid category = {category}/>)
        expect(wrapper).toMatchSnapshot();
     })


     test('debe de mostrar items cuando se cargan imagenes useFetchGifs', ()=>{

        
        const gifs = [{
            id: 'abc',
            url: 'https://localhost/hola.jpg',
            title: 'hola'
        },
        {
            id:'123',
            url: 'https://localhost/adios.jpg',
            title: 'adios'
        }]
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        })
        
        const wrapper = shallow(<GifGrid category={category}/>)

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false)
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
     })
 })