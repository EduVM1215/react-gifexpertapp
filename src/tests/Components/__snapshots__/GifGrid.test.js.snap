// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests sobre GifGrid debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate_animated animate__fadeInDown"
  >
    Metal Gear
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Tests sobre GifGrid debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate_animated animate__fadeInDown"
  >
    Metal Gear
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="hola"
      url="https://localhost/hola.jpg"
    />
    <GifGridItem
      id="123"
      key="123"
      title="adios"
      url="https://localhost/adios.jpg"
    />
  </div>
</Fragment>
`;
